CorsixTH/Lua/announcer.lua

```diff
@@ -137,6 +154,12 @@ function Announcer:playAnnouncement(name, priority, decay_hours, played_callback
   -- It doesn't make sense to play the sacked announcement when the employee
   -- already has had several other jobs and has died already [joke].
 
+  -- Check for duplicate announcements, if there is we refresh the existing one
+  local duplicate_announcement = AnnouncementQueue:checkForDuplicates(self, name)
```

Comment:
Am I missing something, with AnnouncementQueue:checkForDuplicates vs self.entries:checkForDuplicates? You then pass just the game date instead (created_date is assigned below this) of the instance of announcer and obviously adjust checkForDuplicates to process the AnnouncementQueue itself.